### PR TITLE
pcp::types: Specialize `PolyCheckedVector` into `Sum` type

### LINKED ISSUE #94: Naming conventions for `Value` implementations, plus new types
Module `pcp::types` defines the following types:
* `Boolean` -- each input is 1 or 0; the aggregate output is the total.
* `PolyCheckedVector` -- the input is a vector of field elements, where for each element `x`, we have `p(x)==0` for some polynomial `p`. The output is the sum of the input vectors, element-wise. Basically, this is a generalization of a vector of `Boolean`s.
* `MeanVarUnsignedVector` -- the input is a vector of integers in range `[0, 2^n)` for some `n`. The output of the sum of these vectors, as well as the sum of the squares of each element.

We came up with these types in a fairly ad-hoc manner. Basically each demonstrates some aspect of the proof system that we wanted to prototype. Now that we're farther along, I think it's time we settle on the types we actually want to use and to normalize the names.

First, I'd like to suggest the following convention for naming types: The name of the type should reflect the type of the *aggregate* rather than the inputs. For instance, we would rename `Boolean` to `Count`.

Here are the types I think we want to start:
* `Count` -- an 0 or a 1. (This would replace `Boolean`.)
* `CountVec` -- a vector of 0s and 1s. (This would replace `PolyCheckedVector` -- I don't think we have any need for the generalization.)
* `Sum` -- an integer in range `[0,2^n)` for some `n`.
* `SumVec` -- a vector of integers in range `[0, 2^n)` for some `n`.
* `Histogram` -- an entry in a histogram with fixed bucket boundaries. Similar to https://prometheus.io/docs/concepts/metric_types/#histogram.

The reason we want the `Vec` variant of things is that, when we have many instances of the same type, it can be more efficient to encode them all together.

### PR DESCRIPTION
**Pull request by cjpatton**
Based on #110.
Partially closes #94.

### CODE REVIEW

File: src/pcp/types.rs [hunk 1]
```diff
@@ -103,41 +103,58 @@ impl<F: FieldElement> TryFrom<((), &[F])> for Count<F> {
...
+                "value of summand exceeds bit length".to_string(),
+            ));
+        }
+
+        let one = F::Integer::try_from(1).unwrap();
+        let mut data: Vec<F> = Vec::with_capacity(bits);
+        for l in 0..bits {
+            let l = F::Integer::try_from(l).unwrap();
+            let w = F::from((int >> l) & one);
+            data.push(w);
```
Comment (tgeoghegan) [inline src/pcp/types.rs:145]: IIUC, we'll end up encoding a single `Field126` value as 128 `Field126` values, each of which is 0 or 1. Why not represent it as a single `Field126`, and use `poly_range_check(0, Field126::MAX)` as `range_checker`? Or use some caller provided upper bound (perhaps they know something about the data they're recording that lets them provide a tighter upper bound).

**Review by tgeoghegan — CHANGES_REQUESTED** [review]
I think my question here is related to my questions on #109.

File: src/pcp/types.rs (hunk 1 above)
Comment (tgeoghegan) [reply src/pcp/types.rs:145]: OK, that all makes sense. I have no other comments on this PR, but we should resolve #109 so I'll refrain from approving this for now.

**Review by tgeoghegan — APPROVED** [review]

**Comment by cjpatton** [comment]
Removed `PolyCheckedVector` benchmark.

**Pull request merged**